src/k8s/pkg/utils/time.go

```diff
@@ -0,0 +1,51 @@
+package utils
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// SecondsToExpirationDate returns the expiration date from now plus the given
+// number of seconds.
+func SecondsToExpirationDate(now time.Time, seconds int) time.Time {
+	switch {
+	case seconds%(365*24*60*60) == 0:
+		return now.AddDate(seconds/365/24/60/60, 0, 0)
+	case seconds%(30*24*60*60) == 0:
+		return now.AddDate(0, seconds/30/24/60/60, 0)
+	case seconds%(24*60*60) == 0:
+		return now.AddDate(0, 0, seconds/24/60/60)
+	default:
+		return now.Add(time.Duration(seconds) * time.Second)
+	}
+}
+
+// TTLToSeconds converts a TTL string to seconds. The TTL string is a number
+// followed by a unit:
+// - y: years
+// - m: months
+// - d: days
+func TTLToSeconds(ttl string) (int, error) {
```

Comment:
we still need to parse proper durations, e.g. 8640h. maybe fallthrough to `time.ParseDuration`, or support `h` as well